Sources/AsyncAlgorithms/AsyncBufferSequence.swift

```diff
@@ -234,8 +234,8 @@ public struct AsyncBufferSequence<Base: AsyncSequence, Buffer: AsyncBuffer> wher
   }
 }
 
-extension AsyncBufferSequence: Sendable where Base: Sendable, Base.AsyncIterator: Sendable, Base.Element: Sendable { }
-extension AsyncBufferSequence.Iterator: Sendable where Base: Sendable, Base.AsyncIterator: Sendable, Base.Element: Sendable { }
+extension AsyncBufferSequence: Sendable where Base: Sendable { }
```

Comment:
@kperryua do you think we should just make `AsyncBufferSequence` require the `Base` to be `Sendable` and avoid this type of thing?

Reply:
Just to be clear, the warnings are specific to the `Base.Element: Sendable` constraint. I took the liberty to also remove the `Base.AsyncIterator` constraint as well.

The declaration of `AsyncBufferSequence` already defines them:
```swift
public struct AsyncBufferSequence<Base: AsyncSequence, Buffer: AsyncBuffer> where Base.Element == Buffer.Input, Base.AsyncIterator: Sendable {  }
```
- It contains an explicit `AsyncIterator` constraint.
- `Base.Element` must be `Buffer.Input (aka AsyncBuffer.Input)`, which is already `Sendable` (line 130).